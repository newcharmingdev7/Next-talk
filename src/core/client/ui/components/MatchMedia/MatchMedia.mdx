---
name: MatchMedia
menu: UI Kit
---

import { Playground, PropsTable } from 'docz'
import MatchMedia from './MatchMedia'

# MatchMedia

`MatchMedia` uses media queries to provide responsive rendering.

## Basic usage
<Playground>
  <MatchMedia maxWidth="xs">
    I'm a very small screen
  </MatchMedia>
  <MatchMedia minWidth="xs" maxWidth="sm">
    I'm a small screen
  </MatchMedia>
  <MatchMedia minWidth="sm" maxWidth="md">
    I'm a medium screen
  </MatchMedia>
  <MatchMedia minWidth="md" maxWidth="lg">
    I'm a large screen
  </MatchMedia>
  <MatchMedia minWidth="lg" maxWidth="xl">
    I'm a very large screen
  </MatchMedia>
  <MatchMedia minWidth="xl">
    I'm a super large screen
  </MatchMedia>
</Playground>

## Using render props
<Playground>
  <MatchMedia minWidth="lg">
    {matches => <span>{matches ? "I'm big" : "I'm small"}</span>}
  </MatchMedia>
</Playground>
