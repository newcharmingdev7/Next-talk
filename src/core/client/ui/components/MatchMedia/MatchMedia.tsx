import React from "react";
import { ReactNode, StatelessComponent } from "react";
import Responsive from "react-responsive";

import theme from "../../theme/variables";

type Breakpoints = keyof typeof theme.breakpoints;

interface InnerProps {
  minWidth?: Breakpoints;
  maxWidth?: Breakpoints;
  children: ReactNode | ((matches: boolean) => React.ReactNode);
  className?: string;
  component?:
    | string
    | React.SFC<any>
    | React.ClassType<any, any, any>
    | React.ComponentClass<any>;
  all?: boolean;
  print?: boolean;
  screen?: boolean;
  speech?: boolean;
}

const MatchMedia: StatelessComponent<InnerProps> = props => {
  const { speech, minWidth, maxWidth, ...rest } = props;
  const mapped = {
    // TODO: Temporarily map newer speech to older aural type until
    // react-responsive supports the speech prop.
    aural: speech,
    minWidth: minWidth ? theme.breakpoints[minWidth] + 1 : undefined,
    maxWidth: maxWidth ? theme.breakpoints[maxWidth] : undefined,
  };
  return <Responsive {...rest} {...mapped} />;
};

export default MatchMedia;
