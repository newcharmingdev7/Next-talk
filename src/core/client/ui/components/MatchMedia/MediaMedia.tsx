import React from "react";
import { ReactNode, StatelessComponent } from "react";
import Responsive from "react-responsive";

interface InnerProps {
  minWidth?: string;
  maxWidth?: string;
  children: ReactNode;
  className?: string;
  component?:
    | string
    | React.SFC<any>
    | React.ClassType<any, any, any>
    | React.ComponentClass<any>;
  all?: boolean;
  print?: boolean;
  screen?: boolean;
  speech?: boolean;
}

const MatchMedia: StatelessComponent<InnerProps> = props => {
  // TODO: Temporarily map newer speech to older aural type until
  // react-responsive supports the speech prop.
  const { speech, ...rest } = props;
  return <Responsive {...rest} aural={speech} />;
};

export default MatchMedia;
