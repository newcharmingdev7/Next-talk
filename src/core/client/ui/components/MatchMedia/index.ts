export * from "./MatchMedia";
export { default } from "./MatchMedia";
