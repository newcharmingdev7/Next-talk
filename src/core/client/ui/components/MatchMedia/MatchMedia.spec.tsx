import { shallow } from "enzyme";
import React from "react";

import MatchMedia from "./MatchMedia";

it("renders correctly", () => {
  const props = {
    minWidth: "xs",
    maxWidth: "sm",
    component: "div",
    screen: true,
  };
  const wrapper = shallow(
    <MatchMedia {...props}>
      <div>Hello World</div>
    </MatchMedia>
  );
  expect(wrapper).toMatchSnapshot();
});

it("map new speech prop to older aural prop", () => {
  const props = {
    speech: true,
  };
  const wrapper = shallow(
    <MatchMedia {...props}>
      <div>Hello World</div>
    </MatchMedia>
  );
  expect(wrapper).toMatchSnapshot();
});
