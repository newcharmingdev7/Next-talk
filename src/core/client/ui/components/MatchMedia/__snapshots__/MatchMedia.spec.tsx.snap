// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map new speech prop to older aural prop 1`] = `
<MediaQuery
  aural={true}
  values={Object {}}
>
  <div>
    Hello World
  </div>
</MediaQuery>
`;

exports[`renders correctly 1`] = `
<MediaQuery
  component="div"
  maxWidth="sm"
  minWidth="xs"
  screen={true}
  values={Object {}}
>
  <div>
    Hello World
  </div>
</MediaQuery>
`;
